server/src/main/scala/de/frosner/broccoli/instances/storage/filesystem/FileSystemInstanceStorage.scala

```diff
@@ -79,11 +82,13 @@ class FileSystemInstanceStorage(storageDirectory: File) extends InstanceStorage
   @volatile
   override def writeInstanceImpl(instance: Instance): Try[Instance] = {
     val id = instance.id
-    val file = idToFile(id)
+    val file = idToFileTemp(id)
     val printStream = Try(new PrintStream(new FileOutputStream(file)))
     val afterWrite = printStream.map(_.append(Json.toJson(instance).toString()))
-    printStream.map(_.close())
-    afterWrite.map(_ => instance)
+    val finishWrite = printStream.map(_.close())
+    val finishMove = finishWrite.flatMap(_ =>
+      Try(Files.move(idToFileTemp(id).toPath, idToFile(id).toPath, StandardCopyOption.ATOMIC_MOVE)))
```

Comment:
Please reuse the variable you created instead of again generating the file name with `idToFileTemp(id)`.